common/functions/src/scalars/strings/repeat.rs

```diff
@@ -0,0 +1,234 @@
+// Copyright 2021 Datafuse Labs.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::fmt;
+
+use common_datavalues::prelude::*;
+use common_exception::ErrorCode;
+use common_exception::Result;
+
+use crate::scalars::function_factory::FunctionDescription;
+use crate::scalars::function_factory::FunctionFeatures;
+use crate::scalars::Function;
+
+macro_rules! const_array {
+    ($input_string:expr, $times:expr, $input_rows:expr) => {{
+        let mut string_builder = StringArrayBuilder::with_capacity($input_rows);
+        for n in $times {
+            let value = n.and_then(|n| {
+                if *n >= 0 {
+                    Some($input_string.repeat(*n as usize))
```

Comment:
How about adding `max_repeat_times` check.  I don't want this to be an oom killer.

Reply:
https://github.com/ClickHouse/ClickHouse/blob/master/src/Functions/repeat.cpp#L26-L32